plugins/editorconfig/src/org/editorconfig/language/util/Collections.kt

```diff
@@ -0,0 +1,131 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package org.editorconfig.language.util
+
+import com.intellij.util.Consumer
+
+inline fun <reified R> Iterable<*>.ofType(): List<R> {
+  @Suppress("RemoveExplicitTypeArguments")
+  return ofTypeTo(ArrayList<R>())
+}
+
+inline fun <reified R, C : MutableCollection<in R>> Iterable<*>.ofTypeTo(destination: C): C {
+  forEach { (it as? R)?.let(destination::add) }
+  return destination
+}
+
+inline fun <T> Iterable<T>.takeUntil(predicate: (T) -> Boolean): List<T> =
+  takeWhile { !predicate(it) }
+
+inline fun <T> Iterable<T>.dropUntil(predicate: (T) -> Boolean): List<T> =
+  dropWhile { !predicate(it) }
+
+fun <T> Iterable<T>.drop(): List<T> = drop(1)
+
+fun <T> List<T>.dropLast(): List<T> = dropLast(1)
+
+// it would also  be nice to have `inline fun <T> List<T>.dropFirst(predicate: (T) -> Boolean): List<T>`
+
+fun <T> Iterable<T>.addAllTo(destination: Consumer<T>) =
+  forEach(destination::consume)
+
+fun <T> Iterable<T>.allEqual(): Boolean {
+  var started = false
+  var first: T? = null
+  for (it in this) {
+    if (!started) {
+      first = it
+      started = true
+      continue
+    }
+    if (first != it) return false
+  }
+  return true
+}
+
+inline fun <T, K> Iterable<T>.allEqualBy(selector: (T) -> K): Boolean {
+  var started = false
+  var first: K? = null
+  for (it in this) {
+    if (!started) {
+      first = selector(it)
+      started = true
+      continue
+    }
+    if (first != selector(it)) return false
+  }
+  return true
+}
+
+inline fun <reified T> Iterable<T>.previousAwareAll(acceptor: (current: T, previous: T) -> Boolean): Boolean {
+  var started = false
+  var previous: T? = null
+  for (it in this) {
+    if (!started) {
+      previous = it
+      started = true
+      continue
+    }
+    if (!acceptor(it, previous as T)) return false
+    previous = it
+  }
+  return true
+}
+
+inline fun <reified T, R> Iterable<T>.previousAwareForEvery(acceptor: (current: T, previous: T) -> R) {
+  var started = false
+  var previous: T? = null
+  for (it in this) {
+    if (!started) {
+      previous = it
+      started = true
+      continue
+    }
+    acceptor(it, previous as T)
+    previous = it
+  }
+}
+
+//inline fun <reified T> Iterable<*>.castAll(): List<T> {
+//  val result = mutableListOf<T>()
+//  forEach {
+//    result.add(it as T)
+//  }
+//  return result
+//}
+
+inline fun <reified T> Iterable<*>.tryCastAll(): List<T>? {
+  val result = mutableListOf<T>()
+  forEach {
+    if (it is T) result.add(it)
+    else return null
+  }
+  return result
+}
+
+inline fun <T, R> Iterable<T>.forEvery(action: (T) -> R) {
+  forEach { action(it) }
+}
+
+inline fun <T, R : Any> Iterable<T>.firstOrNullAs(action: (T) -> R?): R? {
+  forEach {
+    val result = action(it)
+    if (result != null) return result
+  }
+  return null
+}
+
+@Suppress("unused")
```

Comment:
remove this, kotlin allows you to leave a value without using it. the much worse effect of that will be suggesting of this valu in completion for every type in IDEA source code

Reply:
There are some places where Kotlin syntax strictly demands `Unit` to be left, like in expression-bodied functions, for example. You are right, though, I'll remove this